### PR TITLE
feat: Adding CloudNative-PG operator to manage PostgreSQL on EKS

### LINKED ISSUE #1288: [ADDON] - Add Addons for CloudNative PG operator
Kubernetes Add-on for [CloudNative PG operator](https://cloudnative-pg.io/) deployment in EKS
Example pattern for deploying production grade CloudNative PG deployment in EKS with best practices
Documentation for CloudNative PG usage with few examples

### PR DESCRIPTION
**Pull request by jihed**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

- Resolves #1288

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [x] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by jihed** [comment]
examples will be part of data-on-eks.

File: modules/kubernetes-addons/cloudnative-pg/main.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+locals {
+  name = "cnpg"
+
+  # https://github.com/cloudnative-pg/charts/blob/main/charts/cloudnative-pg/Chart.yaml
+  default_helm_config = {
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/cloudnative-pg/main.tf:5]: move these down to the module declaration - no need to define in a local here

File: modules/kubernetes-addons/main.tf [hunk 2]
```diff
@@ -112,6 +112,13 @@ module "airflow" {
   addon_context = local.addon_context
 }
 
+module "cloudnative_pg" {
+  count         = var.enable_cnpg ? 1 : 0
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/main.tf:116]: this should follow the format 

```hcl
module "foo" {
  source = "x"
  
  count = var.enable ? 1 : 0
  
  <rest of attributes>
}
```

File: modules/kubernetes-addons/variables.tf [hunk 3]
```diff
@@ -1197,6 +1197,19 @@ variable "airflow_helm_config" {
   default     = {}
 }
 
+#-----------CloudNative PG (Postresql) ADDON-------------
+variable "enable_cnpg" {
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/variables.tf:1201]: Carry chart name throughout naming for consistency:

```suggestion
variable "enable_cloudnative_pg" {
```

File: modules/kubernetes-addons/cloudnative-pg/versions.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">= 1.0.0"
+
+  required_providers {
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/cloudnative-pg/versions.tf:4]: There are no providers used in this addon so you can remove lines 3-13

File: modules/kubernetes-addons/cloudnative-pg/values.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Operator configuration
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply modules/kubernetes-addons/cloudnative-pg/values.yaml:1]: Also, if there are no overrides, simply remove this entirely

[the remark this answers is not in the document]
Comment (jihed) [reply modules/kubernetes-addons/cloudnative-pg/values.yaml]: deleted the default values.

File: modules/kubernetes-addons/cloudnative-pg/main.tf (hunk 1 above)
Comment (jihed) [reply modules/kubernetes-addons/cloudnative-pg/main.tf]: we may need it for the future update (like version of the charts)

**Comment by bryantbiggs** [comment]
please see comment left here https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1288#issuecomment-1399263012

closing the PR for now, we can always re-open in the future if we plan to proceed with providing support for this addon

File: modules/kubernetes-addons/cloudnative-pg/values.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+resources:
+  limits:
+    cpu: 100m
+    memory: 200Mi
+  requests:
+    cpu: 100m
+    memory: 100Mi
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/cloudnative-pg/values.yaml:7]: Remove this file. Leave the defaults comes with the Helm Chart. You can pass advanced configuration from Data on EKS Blueprint

File: modules/kubernetes-addons/cloudnative-pg/main.tf [hunk 7]
```diff
@@ -0,0 +1,24 @@
+locals {
+  name = "cnpg"
+  default_helm_config = {
+    name             = local.name
+    chart            = "cloudnative-pg"
+    repository       = "https://cloudnative-pg.github.io/charts"
+    version          = "0.16.1"
+    namespace        = "${local.name}-system"
+    create_namespace = true
+    values           = [templatefile("${path.module}/values.yaml", {})]
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/cloudnative-pg/main.tf:10]: remove this line

File: modules/kubernetes-addons/cloudnative-pg/variables.tf [hunk 8]
```diff
@@ -0,0 +1,22 @@
+variable "helm_config" {
+  description = "Helm provider config for the airflow."
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/cloudnative-pg/variables.tf:2]: Update the description with Cloudnative PG Operator config and remove references to airflow

**Review by vara-bonthu — COMMENTED** [review]
@jihed Please update these comments and reopen the PR

**Pull request closed without merging**